### ISSUE TITLE
Feed is not ovewritten when custom extension is used

### ISSUE THREAD

**Issue by labdmitriy** [scrapy/scrapy#5516]
### Description

I'm trying to export `scrapy crawl` results to JSON Lines format to the file with extension `.jsonl` (this is requirement of the external system in our case) and ovewrite the file for multiple executions.
As I understand, only `.jl` and `.jsonlines` extensions are supported now and `.jsonl` was discussed in #4848 but not supported yet.
So in this case I tried to use `-O` argument with `--output-format` for `scrapy crawl` command.

### Steps to Reproduce

1. `scrapy crawl -O <filename>.jsonl --output-format jl <spider_name>` 
    OR 
    `scrapy crawl -O <filename>.jsonl --output-format jsonlines <spider_name>`

**Expected behavior:** File is ovewritten with parsed content.

**Actual behavior:** Parsed content is appended to the end of existing file.

**Reproduces how often:** 100%

### Versions

Scrapy       : 2.6.1
lxml         : 4.9.0.0
libxml2      : 2.9.14
cssselect    : 1.1.0
parsel       : 1.6.0
w3lib        : 1.22.0
Twisted      : 22.4.0
Python       : 3.10.4 (main, Jun  4 2022, 14:29:37) [GCC 9.4.0]
pyOpenSSL    : 22.0.0 (OpenSSL 3.0.3 3 May 2022)
cryptography : 37.0.2
Platform     : Linux-5.13.0-44-generic-x86_64-with-glibc2.31

### Additional context

If I use `scrapy crawl -O <filename>.jl <spider_name>` or `scrapy crawl -O <filename>.jsonlines <spider_name>`, file is overwritten successfully, but it seems that the case above is expected to have the same behaviour.

**Comment by Gallaecio** [comment]
Oh, this is actually as designed, but there is room for documentation (log message) improvement.

When you use the failing syntax, you get a warning:

> ScrapyDeprecationWarning: The -t command line option is deprecated in favor of specifying the output format within the output URI. See the documentation of the -o and -O options for more information.

`-t` is `--output-format` (this is what can be improved, the message should probably say `-t/--output-format` instead).

So, since the introduction of support for multiple output feeds, you are expected to use this syntax instead:

```
scrapy crawl -O <filename>.jsonl:jsonlines
```

**Comment by labdmitriy** [comment]
Thanks a lot for your feedback, 
Sorry I really didn't notice this warning because of a lot of log messages.

Your approach is working, thank you!

>  See the documentation of the -o and -O options for more information.
I tried to find where this syntax with colon is mentioned in documentation but didn't find any information about it.

**Comment by Gallaecio** [comment]
Yes, I believe that last behavior is also known, although opening a feature request to make it more smart is probably OK, and maybe a good first issue (and those often get implemented quickly).

**Comment by Dyn4mi** [comment]
Hello. I'd like to contribute and try to fix this issue, but I'm having trouble following the conversation.
I've read the thread, but I got confused on what I am supposed to fix here.
I've noticed 3 issues here.  
  
1. The original issue, where parsed content is appended at the end instead of overwriting the output.  
2. Error message on bad syntax does not show anything about "--output-format", and should give some examples too.  
3. Documentation is outdated.  
  
Which of these am I supposed to work in this issue?  
This is the first time I am trying to contribute to open source, sorry if I am being a nuisance for asking basics.

**Comment by Gallaecio** [comment]
The original issue could be addressed either by making `--output-format` work with `-O`, by having the command fail when those are combined, or by making it so that the warning clearly indicates that `-O` is not supported when using `--output-format`. The first option is the most user-friendly, the second will stress the deprecation of `--output-format`, while the last one may be the worse since users can easily miss the warning. But I am personally OK with any of the 3.

Issues 2 and 3 should be (ideally) both be addressed on the same pull request. But if you prefer to focus on 1 issue, feel free.

**Comment by Dyn4mi** [comment]
I get it now. Thank you.
(What would people reply here? I have no idea of what would be good manners here.)

**Comment by Dyn4mi** [comment]
I've noticed that `--output-format` and `-O` are the same option.
Are other options intended to be able to be used multiple times in the same line or should I implement a sanity check to 
stop parsing duplicated options?

**Comment by Gallaecio** [comment]
I don’t think it is a common problem, I think a specific solution is the best approach here.

**Comment by Dyn4mi** [comment]
Hello. Sorry for taking so long, I've had some problems on my end running VSCode.
I've implemented the error + exit in the code, and it only changes the code flow when --output-format and -overwrite-output
is both set.
But I am having some problems fixing the help message and the documentation.
I've tried walking the code step by step, but I can't seem to understand where the help messages comes from.
All I see is functions getting appended to a list, and even if I backtrack where the functions come from I just can't find the
String containing the help message.
I have no idea on how to edit the documentation too.
Thank you for any help.
